airflow/contrib/operators/gcs_to_gcs.py

```diff
@@ -143,51 +143,50 @@ def execute(self, context):
+                self.source_bucket)
+            self.destination_bucket = self.source_bucket
 
         if self.wildcard in self.source_object:
+
+            if self.source_object.count(self.wildcard) > 1:
+                raise Exception("You can only use one wildcard in source_object parameter.")
+
             prefix, delimiter = self.source_object.split(self.wildcard, 1)
             objects = hook.list(self.source_bucket, prefix=prefix, delimiter=delimiter)
 
             for source_object in objects:
-                if self.last_modified_time is not None:
-                    # Check to see if object was modified after last_modified_time
-                    if hook.is_updated_after(self.source_bucket, source_object,
-                                             self.last_modified_time):
-                        pass
-                    else:
-                        continue
                 if self.destination_object is None:
                     destination_object = source_object
                 else:
                     destination_object = source_object.replace(prefix,
                                                                self.destination_object, 1)
-                self.log.info(
-                    log_message.format(self.source_bucket, source_object,
-                                       self.destination_bucket, destination_object)
-                )
-
-                hook.rewrite(self.source_bucket, source_object,
-                             self.destination_bucket, destination_object)
-                if self.move_object:
-                    hook.delete(self.source_bucket, source_object)
 
+                self._copy_single_object(hook=hook, source_object=source_object,
+                                         destination_object=destination_object)
         else:
-            if self.last_modified_time is not None:
-                if hook.is_updated_after(self.source_bucket,
-                                         self.source_object,
-                                         self.last_modified_time):
-                    pass
-                else:
-                    return
-
-            self.log.info(
-                log_message.format(self.source_bucket, self.source_object,
-                                   self.destination_bucket or self.source_bucket,
-                                   self.destination_object or self.source_object)
-            )
-            hook.rewrite(self.source_bucket, self.source_object,
-                         self.destination_bucket, self.destination_object)
-
-            if self.move_object:
-                hook.delete(self.source_bucket, self.source_object)
+            self._copy_single_object(hook=hook, source_object=self.source_object,
+                                     destination_object=self.destination_object)
+
+    def _copy_single_object(self, hook, source_object, destination_object):
+        if self.last_modified_time is not None:
```

Comment:
I think the docstring might be a bit misleading explaining "last_modified_time" behaviour:

"When specified, if the object(s) were modified after last_modified_time, they will be copied/moved. If tzinfo has not been set, UTC will be assumed."

I think something like that could be better:

"When specified, the objects will be copied or moved, only if they were modified after last_modified_time. If tzinfo has not been set, UTC will be assumed."

Reply:
Updated.